src/Containers/App/index.jsx

```diff
@@ -777,33 +778,33 @@ class App extends Component {
     }
 
     try {
-      let apiVersion = await this.serial.getApiVersion().catch((e) => {
-        if (e.message.includes("Timed out after")) {
-          return null;
-        }
-        throw e;
-      });
+      let apiVersion = null;
       
-      if (apiVersion === null) {
-        let hasResets = false;
-        let i = 0;
-
-        while ((await this.serial.getFourWayInterfaceInfo(i).catch(() => null)) !== null) {
-          try {
-            await this.serial.resetFourWayInterface(i);
-            hasResets = true;
-          } catch(e) {
-            this.addLogMessage('resetEscFailed', { index: i + 1 });
+      try {
+        apiVersion = await this.serial.getApiVersion();
+      } catch(e) {
+        if (e instanceof TimeoutError) {
+          let hasResets = false;
+          let i = 0;
+
+          while ((await this.serial.getFourWayInterfaceInfo(i).catch(() => null)) !== null) {
+            try {
+              await this.serial.resetFourWayInterface(i);
+              hasResets = true;
+            } catch(e) {
+              this.addLogMessage('resetEscFailed', { index: i + 1 });
+            }
+            
+            i += 1;
           }
-          
-          i += 1;
-        }
 
-        if (hasResets) {
-          await this.serial.exitFourWayInterface();
-        }
+          if (hasResets) {
+            await this.serial.exitFourWayInterface();
+          }
 
-        apiVersion = await this.serial.getApiVersion();
+          apiVersion = await this.serial.getApiVersion();
+        }
+        throw e;
```

Comment:
Careful - this thorws in any case. If we handled the reset correctly we don't need to throw here but can continue with regular execution, I think you should re-throw in the else path instead.

Reply:
fixed